### ISSUE TITLE
many wine and native linux games break when installed to another drive like an external

### ISSUE THREAD

**Issue by Snuzzums** [sharkwouter/minigalaxy#569]
I think a "junction" when installed to other drives should fix this. I will try to junction the external install and point it to the default install location, but because all the folders for the games change the directory for each game individually, thus messing with where their install/execution scripts point to, I don't think I'll be too successful. I will look into how to write a junction in shell later today. If you guys could work on it when you get the chance that would be cool, as anything I make will just be a band-aid, not a real solution.

Thanks ahead of time,
-Andrew

**Comment by Snuzzums** [comment]
Update: I got the "junction" which is just a symbolic link working, and minigalaxy "thought" my external GOG folder was the one in $home folder, but it didn't fix any of the games that were acting up. I tried, sorry I couldn't be of more help.

-Andrew

**Comment by sharkwouter** [comment]
Thanks for the report @Snuzzums, shame you could not reproduce it. I think there are some issues with installing Windows games when innoextract is not installed. I'm not sure if that is what you were seeing. I hope it's not an issue that is seen by a lot of people.

**Comment by spillner** [comment]
I ran into something similar (although I was not installing to a different drive) with several games, including but not limited to King's Quest IV and Thief II.  Minigalaxy's installation process would run and mark the game as "installed", but the actual game installer never launched and the only thing left in the game directory was the generic WINE prefix.  Other games with near-identical installers (e.g. King's Quest V, Thief I) would work just fine on the same system.

Your comment about innoextract prompted me to upgrade it from version 1.3 to 1.9, and so far all of the previously-failed games now install successfully.  I presume the underlying issue is that old versions of innoextract fail on certain .cab files, but Minigalaxy should be able to detect and handle these failures more gracefully, e.g. by informing the user that innoextract exited abnormally, cleaning up the vanilla prefix/ directory, and not treating the game as installed in the Minigalaxy UI.

**Comment by sharkwouter** [comment]
That's good to know, thanks for the update. I'll keep that in mind for when people bring up similar issues.

**Comment by GB609** [comment]
When we are talking about "external" drives, the filesystem used matters as well, at least for windows games.
Most external drives tend to be formatted in exfat and this does not support symlinks which are used by wine itself as well as the wine installer method. Moving a game onto such a file system will break the links in the dosdevices folder.
There possibly are ways to get it to work using either bind mounts or overlays, but that would require minigalaxy to maintain its own list of links per game to restore them before game start.
The easier way would be to check the file system of the install target dir and show an error when it doesnt support links.
But theres nothing minigalaxy can do if the user manually moves the game to an external drive after installation.

**Comment by Snuzzums** [comment]
God bless you all for helping me so much on this. I found that downloading the windows version of games from the web and then installing them under lutris works the best. I prefer to use ext4, but if using a drive with exfat or ntfs would work better for symbolic links that would be very useful information. Since this effects more than just me, I'll try again and see if I can get it working, if I do, I'll post again.

**Comment by Snuzzums** [comment]
Having the option to download just the installers, and having the option to download the windows version even when there is a linux version available would fix this, I think I was overthinking the problem by using symbolic links.

**Comment by GB609** [comment]
No, symbolic links do not work in exfat at all. Exfat does **not** support them.
But the installer in minigalaxy currently is buggy. Your issues might also be related to that.

**Comment by GB609** [comment]
I think the last request - manually select windows or linux version - is what it actually boils down to. As such, i'm closing this as a duplicate.
